### PR TITLE
rainbowkit

### PR DESCRIPTION
**Pull request by desamtralized**
- Add RainbowKit and Wagmi dependencies
- Refactor WalletButton for simultaneous Bitcoin/Ethereum support

### CODE REVIEW

File: app/walletButton.tsx [hunk 1]
```diff
@@ -9,128 +9,177 @@ import {
...
+
+  const [btcBalance, setBtcBalance] = useState(0);
 
   useEffect(() => {
-    (async () => {
-      setBalance(await wallet.getBalance());
-    })();
-  }, [wallet]);
+    async function fetchBitcoinBalance() {
+      if (wallets.bitcoin) {
```
Comment (MattNguyen) [inline app/walletButton.tsx]: would prefer

```
      if (!wallets.bitcoin) return;
      const balance = await wallets.bitcoin.getBalance();
      setBtcBalance(balance);

File: app/walletButton.tsx [hunk 2]
```diff
@@ -9,128 +9,177 @@ import {
...
+      setWallets(prev => ({
+        ...prev,
+        ethereum: {
+          chain: "ethereum",
+          symbol: "ETH",
+          address: ethAddress,
+          getBalance: async () => Number(ethBalance?.formatted || 0),
+        },
+      }));
+    } else {
```
Comment (MattNguyen) [inline app/walletButton.tsx]: prefer not using an else statement here. add a `return` on the previous line and unindent L81-85

File: app/walletButton.tsx [hunk 3]
```diff
@@ -9,128 +9,177 @@ import {
...
+  useEffect(() => {
+    updateEthereumWallet();
+  }, [updateEthereumWallet]);
+
+  async function onConnectBitcoin() {
+    try {
+      const bitcoinWallet = await bitcoinConnectInjected();
+      setWallets(prev => ({ ...prev, bitcoin: bitcoinWallet }));
+    } catch (error) {
+      console.error("Failed to connect Bitcoin wallet:", error);
```
Comment (MattNguyen) [inline app/walletButton.tsx:97]: we should display something to the user

Comment (desamtralized) [reply app/walletButton.tsx:97]: I've added a task on linear for error handling in general.

Comment (MattNguyen) [reply app/walletButton.tsx:97]: Can you link to ticket?

Comment (desamtralized) [reply app/walletButton.tsx:97]: [YLD-66: App wide error handler with error provider + errormodal](https://linear.app/project-chaos/issue/YLD-66/app-wide-error-handler-with-error-provider-errormodal)

**Review by MattNguyen — APPROVED** [review]

**Pull request merged**